src/foam/dao/FileJournal.js

```diff
@@ -281,6 +285,39 @@ foam.CLASS({
           .toString());
       `
     },
+    {
+      name: 'getUnit',
+      documentation: 'Reads a syntatically meaningful unit from the journal',
+      type: 'String',
+      args: [
+        {
+          name: 'reader',
+          type: 'BufferedReader'
+        }
+      ],
+      javaCode: `
+        try {
+          String line = reader.readLine();
+          if ( getMultiLine() ) {
+            StringBuilder sb = new StringBuilder();
+            line = line.trim();
```

Comment:
Do we want to trim every line?  That will remove indentation from scripts.

Reply:
Sure, this helps us check what the lines end with. Also we are not using any indentation sensitive languages within our service scripts so it shouldn't cause issues with parsing.

Reply:
Its not syntactically sensitive, to the machine, but it is relevant to the developers reading and editing the scripts.  I don't want all scripts in the GUI to lose their indentation, that would be really hurt the usability.